File: pregunta 4/preg4.py
# -*- coding: utf-8 -*-
"""
Created on Mon Nov 29 22:38:23 2021

@author: Miguel
"""
def calculadora (operacion, x, y):
    return operacion (x,y)

suma = lambda x,y: x+y
resta = lambda x,y: x-y
multiplicacion = lambda x,y: x*y
division = lambda x,y: x/y

op = 0
while (op != -1):
    
    print ("1.- Sumar\n2.- Restar\n3.- Multiplicar\n4.- Dividir>")
    op = int(input())
    print ("Numero 1:")
    x = int(input())
    print ("Numero 1:")
    y = int(input())
    if (op == 1 ):
        print ("resultado: ", calculadora(suma, x, y))
    else: 
        if (op == 2):
            print ("resultado: ",calculadora(resta, x, y))
        else: 
            if (op == 3):
                print ("resultado: ",calculadora(multiplicacion, x, y))
            else: 
                if (op == 4):
                    print ("resultado: ",calculadora(division, x, y))
                else : print("NA") 

File: pregunta 4/preg4.hs
getnum::IO Int
getnum = do
  s <- getLine
  return (read s)

calculadora :: (a -> b -> c) -> ((a, b) -> c)
calculadora f = \(x,y) -> f x y

suma = \a b -> a + b
resta = \a b -> a + b
multiplicacion = \a b -> a * b
division = \a b -> a `div` b

main = do
	putStr("1.- Sumar\n2.- Restar\n3.- Multiplicar\n4.- Dividir\n>\n")
	opc <- getnum

	putStr("Numero 1:\n")
	x <- getnum
	putStr("Numero 2:\n")
	y <- getnum

	let resultado=if opc==1 then calculadora(suma)(x,y) else if opc == 2 then calculadora(resta)(x,y) else if opc == 3 then calculadora(multiplicacion)(x,y) else calculadora(division)(x,y)

	putStrLn("Resultado:"++show(resultado)++"\n")
	main
	 